ui-components/src/forms/MultiSelectField.tsx

```diff
@@ -0,0 +1,67 @@
+import React, { useRef, useEffect, useMemo } from "react"
+import AriaAutocomplete, { IAriaAutocompleteOptions } from "aria-autocomplete"
+import { UseFormMethods, RegisterOptions } from "react-hook-form"
+import "./MultiSelectField.scss"
+import { Icon } from "../icons/Icon"
+
+interface MultiSelectFieldProps {
+  name: string
+  placeholder?: string
+  dataSource: string | string[] | any[] | Function | Promise<any[]>
+  register: UseFormMethods["register"]
+  validation?: RegisterOptions
+  setValue: UseFormMethods["setValue"]
+  label?: string
+  id?: string
+  dataTestId?: string
+}
+
+const MultiSelectField = (props: MultiSelectFieldProps) => {
+  const autocompleteRef = useRef<HTMLInputElement>(null)
+
+  const { name, register, setValue } = props
+  register({ name }, props.validation)
+
+  useEffect(() => {
+    if (autocompleteRef.current) {
+      AriaAutocomplete(autocompleteRef.current, {
+        source: props.dataSource,
+        delay: 500, // debounce for a half-second
+        inputClassName: "input",
+        multiple: true,
+        placeholder: props.placeholder,
+        deleteOnBackspace: true,
+        showAllControl: true,
+        cssNameSpace: "multi-select-field",
+        onChange: (selected) => {
+          setValue(
+            name,
+            selected.map((item) => item.value)
+          )
+        },
+      })
+    }
+  }, [autocompleteRef, name, setValue, props.dataSource])
```

Comment:
I'm curious how we handle this in all our components - Without having a check to see if props.dataSource exists, is it possible that this useEffect will get run multiple times, for example, before and after data is fetched? Should there be a check to see it exists before attaching it to the ref? Or is that state handled in the parent component, and it's assumed you'll always have data on the first mount?

Reply:
I think the idea is that either you have data on the first mount, or you supply an async function so the data is loaded in real-time. We should make sure `useEffect` doesn't run more than once, unless there's a major change to those form control details. The `dataSource` prop is definitely required, otherwise I believe the aria-autocomplete control won't work correctly.